### ISSUE TITLE
Allow passing the flag

### ISSUE THREAD

**Issue by rubenwardy** [MT-CTF/capturetheflag#73]

**Comment by AKryukov92** [comment]
Flag can be an item instead on virtual state of player.

**Comment by BrianGaucher** [comment]
And maybe flag can be placed. Allowing another player to take it. And making it visible. To win the player must place or use the flag on their home flag. 
If the flag is placed, it has an ABM removing it after 30 seconds. 
IDEA: Flag also has a "name tag". Only visible by home team.

**Comment by ClobberXD** [comment]
It'd be good to have place-able flags. But a nametag isn't the most elegant solution IMHO. We could have another HUD waypoint which is created on_place, and removed on_pickup.

**Comment by BrianGaucher** [comment]
Yeah, a HUD waypoint sounds more elegant, didn't know that was possible. And the way point must be removed if the ABM removes the flag.

Now if team red picks up their red flag, should they carry it back? Or it should return to their base upon on_pickup. I'd say it returns to the base, given placing the flag to give to another teammate adds an offensive advantage. Forcing the defensive team to carry back adds another offensive advantage. IMHO teleporting the flag would balance the game.

On September 25, 2018 1:36:16 AM EDT, ClobberXD <notifications@github.com> wrote:
>It'd be good to have place-able flags. But a nametag isn't the most
>elegant solution IMHO. We could have another HUD waypoint which is
>created on_place, and removed on_pickup.
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/rubenwardy/capturetheflag/issues/73#issuecomment-424213112

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by rubenwardy** [comment]
Entity-based flags?

**Comment by ClobberXD** [comment]
Sounds good. :+1:

It should ideally be larger than item entities for the sake of visual distinction. We could run the picking-up logic when this entity is punched.

**Comment by farooqkz** [comment]
@LoneWolfHT how about re-opening this?

**Comment by LoneWolfHT** [comment]
This requires a large amount of work, and might end up going in a separate mode
